component/remote/http/http.go

```diff
@@ -0,0 +1,269 @@
+// Package http implements the remote.http component.
+package http
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+	common_config "github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/pkg/build"
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+	"github.com/grafana/agent/pkg/river"
+	prom_config "github.com/prometheus/common/config"
+)
+
+var userAgent = fmt.Sprintf("GrafanaAgent/%s", build.Version)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "remote.http",
+		Args:    Arguments{},
+		Exports: Exports{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments control the remote.http component.
+type Arguments struct {
+	URL           string        `river:"url,attr"`
+	PollFrequency time.Duration `river:"poll_frequency,attr,optional"`
+	PollTimeout   time.Duration `river:"poll_timeout,attr,optional"`
```

Comment:
These are marked as optional, but the default value makes the decoding fail. Was the intention to make them required?

Reply:
Oops, I forgot to apply the defaults